toqito/matrix_ops/vector_to_density_matrix.py

```diff
@@ -0,0 +1,48 @@
+"""Convert row or column vector to density matrix."""
+import numpy as np
+
+
+def vector_to_density_matrix(input_array: np.ndarray) -> np.ndarray:
+    """Convert a given vector to a density matrix or return the density matrix if already given.
+
+    If the input is a vector, this function computes the outer product to form a density matrix.
+    If the input is already a density matrix (square matrix), it returns the matrix as is.
+
+    Examples
+    ==========
+
+    As an example, consider one of the Bell states.
+
+    >>> from toqito.states import bell
+    >>> from toqito.matrix_ops import vector_to_density_matrix
+    >>>
+    >>> vector_to_density_matrix(bell(0))
+    [[0.5 0.  0.  0.5]
+    [0.  0.  0.  0. ]
+    [0.  0.  0.  0. ]
+    [0.5 0.  0.  0.5]]
+
```

Comment:
Do we want to switch the output from a `2D` list to an `array`? The return type in the function definition does not match with the output type in the example. 

Or we can add this to #468

Reply:
Good point. The function returns an `np.ndarray`, so the return type of okay, but the output here in the comment could be corrected (as you suggested). Wrapped in `array()`.